tests/integration/components/sidenote-test.js

```diff
@@ -1,26 +1,161 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { click, render } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
+import waitUntil from '@ember/test-helpers/wait-until';
 
 module('Integration | Component | sidenote', function (hooks) {
   setupRenderingTest(hooks);
 
-  test('it renders', async function (assert) {
-    // Set any properties with this.set('myProperty', 'value');
-    // Handle any actions with this.set('myAction', function(val) { ... });
+  hooks.beforeEach(async function () {
+    const note = {
+      id: 1,
+      offsetY: 100,
+      data: { foo: 'bar' },
+    };
 
-    await render(hbs`<Sidenote />`);
+    this.setProperties({
+      note,
+      onRegister: () => {},
+      onResize: () => {},
+      onSelect: () => {},
+      onUnselect: () => {},
+    });
+  });
+
+  test('it registers sidenote to parent', async function (assert) {
+    assert.expect(4);
+    const done = assert.async();
+
+    this.set('onRegister', ({ element, id, offsetY, data }) => {
+      assert.equal(this.note.id, id, 'onRegister receives correct id');
+      assert.equal(
+        this.note.offsetY,
+        offsetY,
+        'onRegister receives correct offsetY'
+      );
+      assert.equal(this.note.data, data, 'onRegister receives correct data');
+      assert.ok(element, 'onRegister receives a truthy element');
+
+      done();
+    });
+
+    await render(hbs`
+      <Sidenote
+        @id={{this.note.id}}
+        @offsetY={{this.note.offsetY}}
+        @data={{this.note.data}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+      />
+    `);
+  });
+
+  test('it expose isActive with correct value', async function (assert) {
+    this.set('selectedSidenoteId', this.note.id);
+
+    await render(hbs`
+      <Sidenote
+        @id={{this.note.id}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+        @selectedSidenoteId={{this.selectedSidenoteId}}
+        as |s|
+      >
+        {{#if s.isActive}}
+          <div data-active-sidenote>This is some dummy text.</div>
+        {{/if}}
+      </Sidenote>
+    `);
+
+    assert.dom('[data-active-sidenote]').exists('isActive is exposed as true');
+
+    this.set('selectedSidenoteId', -1);
+
+    assert
+      .dom('[data-active-sidenote]')
+      .doesNotExist('isActive is exposed as false');
+  });
+
+  test('it triggers onResize when element size change', async function (assert) {
+    this.setProperties({
+      showMore: false,
+      onResizeCalled: false,
```

Comment:
I use these types of boolean (`myFunctionCalled`) in both test files (sidenote & sidenote-wrapper-test). I did not find any better/cleaner solution to do so, feel free to propose anything!